driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java

```diff
@@ -107,16 +109,43 @@ public ChangeStreamDocument(final BsonDocument resumeToken,
      *
      * @since 3.8
      */
-    @BsonCreator
+    @Deprecated
     public ChangeStreamDocument(@BsonProperty("resumeToken") final BsonDocument resumeToken,
                                 @BsonProperty("ns") final BsonDocument namespaceDocument,
                                 @BsonProperty("fullDocument") final TDocument fullDocument,
                                 @BsonProperty("documentKey") final BsonDocument documentKey,
                                 @Nullable @BsonProperty("clusterTime") final BsonTimestamp clusterTime,
                                 @BsonProperty("operationType") final OperationType operationType,
                                 @BsonProperty("updateDescription") final UpdateDescription updateDescription) {
+        this(resumeToken, namespaceDocument, null, fullDocument, documentKey, clusterTime, operationType, updateDescription);
+    }
+
+    /**
+     * Creates a new instance
+     *
+     * @param resumeToken the resume token
+     * @param namespaceDocument the BsonDocument representing the namespace
+     * @param destinationNamespaceDocument the BsonDocument representing the destinatation namespace
+     * @param fullDocument the full document
+     * @param documentKey a document containing the _id of the changed document
+     * @param clusterTime the cluster time at which the change occured
+     * @param operationType the operation type
+     * @param updateDescription the update description
+     *
+     * @since 3.11
+     */
+    @BsonCreator
+    public ChangeStreamDocument(@BsonProperty("resumeToken") final BsonDocument resumeToken,
+                                @BsonProperty("ns") final BsonDocument namespaceDocument,
+                                @Nullable @BsonProperty("to") final BsonDocument destinationNamespaceDocument,
+                                @BsonProperty("fullDocument") final TDocument fullDocument,
+                                @Nullable @BsonProperty("documentKey") final BsonDocument documentKey,
```

Comment:
Turns out more of these are nullable

Reply:
Should all of them to be nullable apart from resumeToken?

Reply:
`fullDocument` should.  I think the rest are properly non-nullable.

Reply:
Depends what the user does with the pipeline?

Reply:
I'm comfortable with ChangeStreamDocument representing a pipeline whose structure hasn't been modified.  If they add fields, for example, ChangeStreamDocument is also inappropriate and they should decode to something else.

But open to other opinions on this too.

Reply:
I made everything null except resumeToken and operationType